pkgs/test/test/runner/set_up_all_test.dart

```diff
@@ -95,4 +95,19 @@ void main() {
     expect(test.stdout, neverEmits(contains('(setUpAll)')));
     await test.shouldExit(0);
   });
+
+  test('does not fail with asserts enabled - issue #2498', () async {
```

Comment:
I'm not sure why no existing tests triggered this, but did trigger when we didn't have the fix in `onPlatform`, but I added specific tests for these referencing the issue.